x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_status_filter.tsx

```diff
@@ -34,45 +40,59 @@ export const RuleStatusFilter = (props: RuleStatusFilterProps) => {
 
   const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
 
-  const onFilterItemClick = useCallback(
-    (newOption: RuleStatus) => () => {
-      if (selectedStatuses.includes(newOption)) {
-        onChange(selectedStatuses.filter((option) => option !== newOption));
-        return;
-      }
-      onChange([...selectedStatuses, newOption]);
+  const onFilterItemChange: EuiSelectableProps['onChange'] = useCallback(
+    async (newOptions: EuiSelectableOption[]) => {
+      const selected = newOptions
+        .filter(({ checked, key }) => checked && key)
```

Comment:
do we need check `key`? seems odd that `key` could be falsy?

also probably better to check `checked === 'on'`, more readable IMO